src/libraries/System.Private.CoreLib/src/System/SearchValues/ProbabilisticMap.cs

```diff
@@ -670,6 +856,108 @@ private static bool TryFindMatchOverlapped<TUseFastContains>(ref char cur, int s
             return false;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryFindLastMatch<TUseFastContains>(ref char cur, uint mask, ref ProbabilisticMapState state, out int index)
+            where TUseFastContains : struct, SearchValues.IRuntimeConst
+        {
+            do
+            {
+                index = 31 - BitOperations.LeadingZeroCount(mask);
+
+                if (state.ConfirmProbabilisticMatch<TUseFastContains>(Unsafe.Add(ref cur, index)))
+                {
+                    return true;
+                }
+
+                // Clear the highest set bit
+                mask = BitOperations.FlipBit(mask, index);
+            }
+            while (mask != 0);
+
+            index = 0;
+            return false;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryFindLastMatchOverlapped<TUseFastContains>(ref char cur, int searchSpaceLength, uint mask, ref ProbabilisticMapState state, out int index)
+            where TUseFastContains : struct, SearchValues.IRuntimeConst
+        {
+            do
+            {
+                index = 31 - BitOperations.LeadingZeroCount(mask);
+
+                // Clear the highest set bit
+                mask = BitOperations.FlipBit(mask, index);
+
+                if (index >= Vector256<ushort>.Count)
+                {
+                    // The potential match is in the second vector.
+                    // Fixup the index to account for how we loaded the second overlapped vector.
+                    index += searchSpaceLength - (2 * Vector256<ushort>.Count);
+                }
+
+                if (state.ConfirmProbabilisticMatch<TUseFastContains>(Unsafe.Add(ref cur, index)))
+                {
+                    return true;
+                }
+            }
+            while (mask != 0);
+
+            index = 0;
+            return false;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryFindLastMatch<TUseFastContains>(ref char cur, ulong mask, ref ProbabilisticMapState state, out int index)
+            where TUseFastContains : struct, SearchValues.IRuntimeConst
+        {
+            do
+            {
+                index = 63 - BitOperations.LeadingZeroCount(mask);
+
+                if (state.ConfirmProbabilisticMatch<TUseFastContains>(Unsafe.Add(ref cur, index)))
+                {
+                    return true;
+                }
+
+                // Clear the highest set bit
+                mask = BitOperations.FlipBit(mask, index);
+            }
+            while (mask != 0);
+
+            index = 0;
+            return false;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryFindLastMatchOverlapped<TUseFastContains>(ref char cur, int searchSpaceLength, ulong mask, ref ProbabilisticMapState state, out int index)
```

Comment:
Should the name of these note they're intended to be used only from the AVX512 implementation?